spring/jta/src/main/java/org/jboss/narayana/quickstart/spring/config/H2DataSource.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2013, Red Hat, Inc. and/or its affiliates, and individual
+ * contributors by the @authors tag. See the copyright.txt in the
+ * distribution for a full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.narayana.quickstart.spring.config;
+
+import com.arjuna.ats.internal.jdbc.DynamicClass;
+import org.h2.jdbcx.JdbcDataSource;
+
+import javax.sql.XADataSource;
+import java.sql.SQLException;
+
+public class H2DataSource implements DynamicClass {
+    @Override
+    public XADataSource getDataSource(String dbName) throws SQLException {
+        JdbcDataSource ds = new JdbcDataSource();
+        ds.setURL("jdbc:h2:file:target/test.db;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE");
+        ds.setUser("sa");
+        ds.setPassword("");
+
```

Comment:
I think it don't need to set the username and password. They should be set in the properties.